client/lib/domains/check-domain-availability.js

```diff
@@ -10,14 +10,20 @@ export function checkDomainAvailability( params, onComplete ) {
 		return;
 	}
 
-	wpcom
-		.undocumented()
-		.isDomainAvailable( domainName, blogId, isCartPreCheck, function ( serverError, result ) {
+	wpcom.req.get(
+		`/domains/${ encodeURIComponent( domainName ) }/is-available`,
+		{
+			blog_id: blogId,
+			apiVersion: '1.3',
+			is_cart_pre_check: isCartPreCheck,
+		},
+		function ( serverError, result ) {
```

Comment:
We could use the opportunity to refactor this to use the more natural Promise `.then()` and `.catch()` instead of using the callback argument. [Here's an example](https://github.com/Automattic/wp-calypso/pull/58054/files#diff-f52170fb5fe49527b6bf72a59ff96afb272fd458711097e715f99515df893c7eL113) of how we did that in a recent PR.

Reply:
Nice, thanks for the example. I'v migrated to `.then()` and `.catch()` in 4af8bf7, let me know what you think!